checker.go

```diff
@@ -8,19 +8,27 @@ import (
 )
 
 // Checker is a fake implementation.
-type Checker struct{}
+type Checker struct {
+	zeroLinger bool
+}
 
-// NewChecker creates a Checker, parameters are ignored.
-func NewChecker(zeroLinger bool) *Checker { return &Checker{} }
+// NewChecker creates a Checker with linger set to zero or not.
+func NewChecker(zeroLinger bool) *Checker {
+	return &Checker{zeroLinger: zeroLinger}
+}
 
 // InitChecker is unnecessary on this platform.
 func (s *Checker) InitChecker() error { return nil }
 
 // CheckAddr performs a TCP check with given TCP address and timeout.
-// NOTE: zeroLinger is ignored on this platform.
+// NOTE: zeroLinger may be ignored on this platform.
```

Comment:
Could you add a more detailed explanation of it? Like when and why it will and will not work.

Reply:
I added the explanation for why `zeroLinger` only works on POSIX operating systems (for now).